data_diff/cloud/datafold_api.py

```diff
@@ -0,0 +1,256 @@
+
+
+class TSummaryResultValueStats(pydantic.BaseModel):
+    total_rows: int
+    rows_with_differences: int
+    total_values: int
+    compared_columns: int
+    columns_with_differences: int
+    columns_diff_stats: List[TSummaryResultColumnDiffStats]
+
+
+class TSummaryResultSchemaStats(pydantic.BaseModel):
+    columns_mismatched: Tuple[int, int]
+    column_type_mismatches: int
+    column_reorders: int
+    column_counts: Tuple[int, int]
+
+
+class TCloudApiDataDiffSummaryResult(pydantic.BaseModel):
+    status: str
+    pks: Optional[TSummaryResultPrimaryKeyStats]
+    values: Optional[TSummaryResultValueStats]
+    schema_: Optional[TSummaryResultSchemaStats]
+    dependencies: Optional[Dict[str, Any]]
+
+    @classmethod
+    def from_orm(cls: Type[Self], obj: Any) -> Self:
+        pks = TSummaryResultPrimaryKeyStats(**obj["pks"]) if "pks" in obj else None
+        values = TSummaryResultValueStats(**obj["values"]) if "values" in obj else None
+        deps = obj["deps"] if "deps" in obj else None
+        schema = TSummaryResultSchemaStats(**obj["schema"]) if "schema" in obj else None
+        return cls(
+            status=obj["status"],
+            pks=pks,
+            values=values,
+            schema_=schema,
+            deps=deps,
+        )
+
+
+class TCloudDataSourceTestResult(pydantic.BaseModel):
+    status: TestDataSourceStatus
+    message: str
+    outcome: str
+
+
+class TCloudApiDataSourceTestResult(pydantic.BaseModel):
+    name: str
+    status: str
+    result: TCloudDataSourceTestResult
+
+
+@dataclasses.dataclass
+class DatafoldAPI:
+    api_key: str
+    host: str = "https://app.datafold.com"
+    timeout: int = 30
+
+    def __post_init__(self):
+        self.host = self.host.rstrip("/")
+        self.headers = {
+            "Authorization": f"Key {self.api_key}",
+            "Content-Type": "application/json",
+        }
+
+    def make_get_request(self, url: str) -> Any:
+        rv = requests.get(url=f"{self.host}/{url}", headers=self.headers, timeout=self.timeout)
+        rv.raise_for_status()
+        return rv
+
+    def make_post_request(self, url: str, payload: Any) -> Any:
+        rv = requests.post(url=f"{self.host}/{url}", headers=self.headers, json=payload, timeout=self.timeout)
+        rv.raise_for_status()
+        return rv
+
+    def get_data_sources(self) -> List[TCloudApiDataSource]:
+        rv = self.make_get_request(url="api/data_sources")
+        rv.raise_for_status()
+        return [TCloudApiDataSource(**item) for item in rv.json()]
+
+    def create_data_source(self, config: TDsConfig) -> TCloudApiDataSource:
+        rv = self.make_post_request(url="api/v1/data_sources", payload=config.dict())
+        return TCloudApiDataSource(**rv.json())
+
+    def get_data_source_schema_config(
+        self,
+        only_important_properties: bool = False,
+    ) -> List[TCloudApiDataSourceConfigSchema]:
+        rv = self.make_get_request(url="api/v1/data_sources/types")
```

Comment:
I'm getting a 401 here:
```
Running with data-diff=0.6.0

Cloud datafold host: https://app.datafold.com
Data source ID not found in dbt_project.yml
Would you like to create a new data source? [y/n]: y
[13:47:42] ERROR - 401 Client Error: Unauthorized for url: https://app.datafold.com/api/v1/data_sources/types
```

Reply:
It is ok. We have not exposed some internal endpoints on SaaS yet but the corresponding PR is ready.

Reply:
ah ok I thought that might be the case